### PR TITLE
Add the package systemtap-sdt-dev to requirements

### LINKED ISSUE #946: missing software-defined tracepoints
A colleague has been trying to use `splat` (https://github.com/open-power-sdk/splat) with an application compiled and linked with Advance Toolchain.  `splat` is a wrapper around the `perf` command which makes use of "software defined tracepoints" (SDTs).  SDTs are predefined tracepoints in user-level/library code which can be enabled and used to do system-wide tracing on user-mode symbols.

Glibc has SDTs defined for many things, including what `splat` uses:  various `pthread_mutex` function calls.  I'm not an expert here, but from what I've found (looking at the code for the `perf` command, the SDT definitions show up in an ELF "note" section in the library.  Some evidence to support that:
```
$ readelf --notes /lib64/libpthread-2.17.so | grep sdt | wc -l
115
```
 
I'm not seeing the same information in the corresponding Advance Toolchain library:
```
$ readelf --notes /opt/at12.0/lib64/libpthread-2.28.so | grep sdt | wc -l
0
```
 
It looks like, in this case, glibc's `configure` script will enable the generation of SDTs by looking for `<sys/sdt.h>` and successfully compiling a small test case with it.  The script ensures that `USE_STAP_PROBE` is defined for the glibc build if that test case works.  `USE_STAP_PROBE` is then used in glibc's `include/stap-probe.h`, which defines `LIBC_PROBE`, which is the macro used to generate these various SDTs, as can be seen in glibc's `nptl/pthread_mutex_init.c`, for example.

I think there is a simple fix, and that is to make sure the build systems have the packages installed that include `sys/sdt.h`.
On Red Hat:  `systemtap-sdt-devel`
On Ubuntu: `systemtap-sdt-dev`
 
I tried to figure out where a dependency should be added, to avoid this in the future, and I think it would be in `configs/*/distros/*`, but I want to confirm because I am not sure.

### PR DESCRIPTION
**Pull request by er-1**
Added the package to requirement on AT 11.0, 12.0 and next.
Enabled systemptap when building glibc.

Fix #946 
    
Signed-off-by: Erwan Prioul <erwan@linux.ibm.com>

### CODE REVIEW

**Comment by er-1** [comment]
AT 12.0 is expecting to fail on Ubuntu18.

**Comment by tuliom** [comment]
Is the at9.0 change intentional? I'm asking because you replaced the symlink on at11.0.

**Comment by er-1** [comment]
> Is the at9.0 change intentional?

That's a mistake.

**Comment by er-1** [comment]
> > Is the at9.0 change intentional?

> That's a mistake.

@tuliom it's fixed.

**Comment by er-1** [comment]
@tuliom I added a new testcase.

**Comment by ThinkOpenly** [comment]
I'm able to add the probes I want using Erwan's build (AT-next on Ubuntu 18.04).

Tested-by: Paul A. Clarke <pc@us.ibm.com>

File: fvtr/systemtap/systemtap.exp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    printit "Skipping: systemtap isn't enabled on version older than AT 11.0"
+    exit $ENOSYS
+}
+set lib "lib64"
+if { $TARGET32 } {
+    set lib "lib"
+}
+set readelf_path [exec which readelf]
+set sdt [exec $readelf_path --notes  "$env(AT_DEST)/$lib/libpthread.so" | \
+	     grep sdt | wc -l]
```
Comment (tuliom) [inline fvtr/systemtap/systemtap.exp]: Don't you need to use `eval` here, e.g. https://github.com/advancetoolchain/advance-toolchain/blob/master/fvtr/shared.exp#L402

Comment (gftg85) [reply fvtr/systemtap/systemtap.exp]: Also, 'sdt' is a simple string, are you sure this won't match anything else?  Maybe replace it with stapsdt?

Comment (er-1) [reply fvtr/systemtap/systemtap.exp]: @gftg85 I replaced `sdt` with `stapsdt`.

File: fvtr/systemtap/systemtap.exp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+if { $env(AT_MAJOR_VERSION) < 11.0 } {
+    printit "Skipping: systemtap isn't enabled on version older than AT 11.0"
+    exit $ENOSYS
+}
+set lib "lib64"
+if { $TARGET32 } {
+    set lib "lib"
+}
+set readelf_path [exec which readelf]
```
[the remark this answers is not in the document]
Comment (er-1) [reply fvtr/systemtap/systemtap.exp]: @tuliom I updated the code with the referenced one.

**Comment by er-1** [comment]
@gftg85 Good point. I followed your hint and updated the patch.

**Pull request merged**